[Guides] Refactor Tiered Prefix Cache (#1345)
llm-d/llm-d: guides/tiered-prefix-cache/storage/README.md

Reviewer: ```suggestion
curl -s http://${IP}:8081/v1/completions  \
```

* **High throughput via parallelism** - I/O operations are parallelized across multiple threads to increase bandwidth and reduce tail latency.
* **Minimal GPU compute interference** - Uses GPU DMA for data transfers, reducing interference with GPU compute kernels during load and store operations.
  ```bash
    export branch="main" # branch, tag, or commit hash
    git clone https://github.com/llm-d/llm-d.git && cd llm-d && git checkout ${branch}
  ```

**Note:** The storage connector does not handle cleanup or eviction of data on the shared storage. Storage capacity management must be handled by the underlying storage system or by an external controller. A simple reference implementation of a PVC-based evictor is available in the [kv-cache repository (PVC Evictor)](https://github.com/llm-d/llm-d-kv-cache), which can be used to automatically clean up old KV-cache files when storage thresholds are exceeded.
- Set the following environment variables:
  ```bash
    export GAIE_VERSION=v1.4.0
    export GUIDE_NAME="tiered-prefix-cache-storage"
    export NAMESPACE=llm-d-${GUIDE_NAME}
  ```
- Install the Gateway API Inference Extension CRDs:

> For advanced configuration options and implementation details, see the [llm-d FS backend documentation](https://github.com/llm-d/llm-d-kv-cache/tree/main/kv_connectors/llmd_fs_backend).

Reviewer: Can you add a link for additional details and configuration, similar to this? Maybe you can put in the "Supported Connectors" table

> export STORAGE_CLASS=default # options: default, lustre, efs-sc

Reviewer: ```suggestion
export STORAGE_CLASS="" # leave empty to use the cluster default StorageClass; or set "lustre" / "efs-sc"
```

    -f guides/recipes/scheduler/base.values.yaml \
    -f guides/tiered-prefix-cache/storage/scheduler/${GUIDE_NAME}.values.yaml \
    -n ${NAMESPACE} --version ${GAIE_VERSION}
```
export STORAGE_CLASS=default
```

No additional provision steps are required.

<!-- TAB:GCP Lustre -->
<details>
<summary><h4>Gateway Mode</h4></summary>

#### GCP Lustre
To use a Kubernetes Gateway managed proxy instead of standalone:

Set your storage class which will be used later to provision the PVC.
1. _Deploy a Kubernetes Gateway_ by following one of [the gateway guides](../../prereq/gateways).
2. _Deploy HTTPRoute and the inference scheduler_:

```bash
export PROVIDER_NAME=gke # options: none, gke, agentgateway, istio
helm install llm-d-infpool \
    oci://registry.k8s.io/gateway-api-inference-extension/charts/inferencepool  \
    -f guides/recipes/scheduler/base.values.yaml \
    --set provider.name=${PROVIDER_NAME} \
    --set experimentalHttpRoute.enabled=true \
    --set experimentalHttpRoute.inferenceGatewayName=llm-d-inference-gateway \
    -n ${NAMESPACE} --version ${GAIE_VERSION}
```
export STORAGE_CLASS=lustre
```

> To provision a managed GCP Lustre instance on GKE and configure the corresponding `StorageClass`, follow the [GCP Lustre guide](./manifests/backends/lustre/README.md).

Reviewer: I think we should keep the instructions on how to set storage class in different vendors, like in EFS or GCP Lustre

> curl -X POST http://${IP}/v1/completions \

Reviewer: ```suggestion
curl -X POST http://${IP}:8081/v1/completions \
```
